helpers/models/common.py

```diff
@@ -912,8 +914,11 @@ def _load_pipeline(
             pipeline_kwargs["watermarker"] = None
         if "watermark" in signature.parameters:
             pipeline_kwargs["watermark"] = None
-        if load_base_model:
-            pipeline_kwargs[self.MODEL_TYPE.value] = self.unwrap_model(model=self.model)
+        if load_base_model and self.model is not None:
+            if self.unwrap_model(model=self.model) is None:
+                pipeline_kwargs[self.MODEL_TYPE.value] = self.model
+            else:
+                pipeline_kwargs[self.MODEL_TYPE.value] = self.unwrap_model(model=self.model)
```

Comment:
this also seems a bit round-about. what's the problem being solved?

Reply:
same as above